pkg/reconciler/taskrun/validation.go

```diff
@@ -85,3 +102,229 @@ func extractPlatform(tr *tektonapi.TaskRun) (string, error) {
 	}
 	return "", errMissingPlatformParameter
 }
+
+// validateNonZeroPositiveNumber validates a string represents a valid integer within the specified range
+// Validation rules:
+// - Must be a valid integer parseable by strconv.Atoi
+// - Must be between 1 and maxValue inclusive
+//
+// Parameters:
+// - value: The string value to validate
+// - maxValue: The maximum allowed value
+//
+// Returns:
+// - int: The validated integer value
+// - error: If value is invalid or out of range
+func validateNonZeroPositiveNumber(value string, maxValue int) (int, error) {
+	num, err := strconv.Atoi(value)
+	if err != nil {
+		return -1, fmt.Errorf("invalid value '%s': must be a valid integer between 1 and %d: %w", value, maxValue, err)
+	}
+	if num < 1 || num > maxValue {
+		return -1, fmt.Errorf("invalid value '%s': must be a valid integer between 1 and %d", value, maxValue)
+	}
+	return num, nil
+}
+
+// validateMaxInstances validates a string represents a valid max-instances value
+// Uses the global maxInstancesValue as the upper limit
+//
+// Parameters:
+// - value: The string value to validate
+//
+// Returns:
+// - int: The validated integer value
+// - error: If value is invalid or out of range
+func validateMaxInstances(value string) (int, error) {
+	result, err := validateNonZeroPositiveNumber(value, maxInstancesValue)
+	if err != nil {
+		return -1, fmt.Errorf("invalid max-instances: %w", err)
+	}
+	return result, nil
+}
```

Comment:
with this check, we'll not be able to scale above `maxInstancesValue` (currently `250`) without having to release a new version. To me we should validate the syntax (e.g. not an number, non-positive number, ...) as we do with `validateNonZeroPositiveNumber`, but not the actual value.

Is there any specific reason for us to have a hard limit to `250`? Should we move this value to configuration?